mlrun/model_monitoring/db/stores/v3io_kv/kv_store.py

```diff
@@ -285,7 +285,23 @@ def list_model_endpoints(
             endpoint = self.get_model_endpoint(
                 endpoint_id=endpoint_id,
             )
-            endpoint_list.append(endpoint)
+            if labels:
+                label_in = True
+                endpoint_labels = json.loads(endpoint["labels"])
+                for label in labels:
+                    if "=" in label:
+                        label, value = list(map(lambda x: x.strip(), label.split("=")))
+                    else:
+                        value = None
+                    label_in = label in endpoint_labels and (
+                        (value and str(endpoint_labels[label]) == value) or not value
+                    )
```

Comment:
I think this can be simplified a bit
```suggestion
                    label_in = label in endpoint_labels and (
                        not value or str(endpoint_labels[label]) == value
                    )
```